### PR TITLE
[Recycle Bin] Optimize relations serialization with ElementDescriptor

### LINKED ISSUE #5426: [Recycle Bin] Use ElementDescriptor to store relations
Currently the entire objects get serialized. 
Similar solution to what we do with versions: 
https://github.com/pimcore/pimcore/blob/1d54158b50d52c3f220d68fb1f667acc3cc99a4b/models/Version.php#L273

Using: `\Pimcore\Model\Version\ElementDescriptor`

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #5426

### CODE REVIEW

File: models/Element/Recyclebin/Item.php [hunk 1]
```diff
@@ -329,6 +325,61 @@ protected function doRecursiveRestore(Element\ElementInterface $element)
...
+     * @return mixed
+     */
+    public function unmarshalData($data)
+    {
+        $copier = new DeepCopy();
+        $copier->addTypeFilter(
+            new \DeepCopy\TypeFilter\ReplaceFilter(
+                function ($currentValue) {
+                    if ($currentValue instanceof Model\Version\ElementDescriptor) {
+                        $value = Element\Service::getElementById($currentValue->getType(), $currentValue->getId());
```
Comment (weisswurstkanone) [inline models/Element/Recyclebin/Item.php]: do we need a check if destination still exists ?

Comment (dvesh3) [reply models/Element/Recyclebin/Item.php]: handled already!

**Comment by brusch** [comment]
@dvesh3 had to revert this PR because there are some troubles with recycle bin items containing children.

**Comment by brusch** [comment]
I guess we need some kind of an extra structure in the serialized data to save children :|

**Pull request merged**